TripalCultivate/TripalCultivate
# Update standards in this module

**Issue #29**

## Motivation

This started as a simple PR to get the base module up to the same repo standards that we are striving for in phenotypes but a number of other issues have cropped up.

## What does this PR do?

- [x] Add issue + PR templates to this repo
- [x] Enable coding standards as currently being initiated in https://github.com/TripalCultivate/TripalCultivate-Phenotypes/pull/119
- [x] Update existing code in this module to match standards (small task since there isn't much code yet)
- [x] Solve how to modify the Drupal standards slightly to match what we want to accomplish for our repos
- [x] Update module to handle new changes to Tripal core
   - [x] URL Alias now added as a base property --increases number of fields to expect
   - [x] New publishing change is causing schema errors --solved by https://github.com/tripal/tripal/pull/2026
   - [x] Update Tripal Content Type and add Tripal Content listings to theme our categories.


## Testing

### Automated Testing

There is no new automated testing for this PR; however, the following changes were made:

- Update the number of fields to add to our assertions for field count since Tripal core has added a new path base property + field.
- Update the files to match coding standards from phpcs
- Change the `$connection property` to `$chado_connection` as we now prefer to refer to it.

### Manual Testing

1. Create a new docker on this branch.
2. Go to Top Bar > Page Structure and ensure that our categories (Research Management/Experiment) match Tripal core.

| Dark | Light |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/e51e3887-16cf-46db-88aa-b695e4daa546) | ![image](https://github.com/user-attachments/assets/7dd01f4b-6791-4c17-a593-03deba3e8a16) |

3. Go to Top Bar > Tripal > Content, click on "Add Tripal Content" and ensure that again, our categories (Research Management/Experiment) match Tripal core.

| Dark | Light |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/061b4152-0115-4671-8512-b0759bf1194b) | ![image](https://github.com/user-attachments/assets/fdf81c3d-c580-46e3-85af-28401e8392da) |
